src/utils/sanitise.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * @notice copied from https://gist.github.com/simonrenoult/2fa79fec8f035b2d35cef0767363b227
+ * */
+
+import _ from 'lodash';
+
+export const sanitize = (object: object): object | string | null => {
+  if (_.isString(object)) return _sanitizeString(object);
+  if (_.isArray(object)) return _sanitizeArray(object);
+  if (_.isPlainObject(object)) return _sanitizeObject(object);
+  return object;
+};
+
+const _sanitizeString = (string: string) => {
+  return _.isEmpty(string) ? null : string;
+};
+
+const _sanitizeArray = (array: Array<any>) => {
+  return _.filter(_.map(array, sanitize), _isProvided);
+};
+
+const _sanitizeObject = (object: object) => {
+  return _.pickBy(_.mapValues(object, sanitize), _isProvided);
+};
+
+const _isProvided = (value: any) => {
+  const typeIsNotSupported = !_.isNull(value) && !_.isString(value) && !_.isArray(value) && !_.isPlainObject(value);
```

Comment:
Hold on, aren't numbers or booleans supported?

Reply:
Good catch!

Reply:
So, after executing `sanitise` with:

```
 sanitise({
    aTextNumber: '123',
    aTrueBool: true,
    aFalseBool: false,
    aTextFalseBool: 'false',
    aTextTrueBool: 'true',
    aNumber: 123,
    aTrueNumber: 1,
    aFalseNumber: 0,
    anUndefined: undefined,
    aNull: null,
    aTextUndefined: "undefined",
    aTextNull: "null"
  }),
```

I'm getting:
```
aFalseBool: false
aFalseNumber: 0
aNumber: 123
aTextFalseBool: "false"
aTextNull: "null"
aTextNumber: "123"
aTextTrueBool: "true"
aTextUndefined: "undefined"
aTrueBool: true
aTrueNumber: 1
```

Which is expected behaviour.

`isEmpty` will yield true for any number or bool.